File: src/main/scala/org/zezutom/puzzle/scala/FizzBuzz.scala
package org.zezutom.puzzle.scala

/**
 * Write a program that prints the numbers from 1 to 100 following the rules below:
 * - for multiples of three print “Fizz” (instead of the number)
 * - for the multiples of five print “Buzz”
 * - for numbers which are multiples of both three and five print “FizzBuzz”
 * - otherwise print the number
 */
class FizzBuzz {

  def fizzBuzz(count:Int):String = {
    Array.tabulate(count)(n => n + 1 match {
      case x if (x % 3 == 0 && x % 5 == 0) => "FizzBuzz"
      case x if (x % 3 == 0) => "Fizz"
      case x if (x % 5 == 0) => "Buzz"
      case _ => (n + 1).toString
    }).mkString(" ").trim()
  }
}

File: src/main/java/org/zezutom/puzzle/java/FizzBuzz.java
package org.zezutom.puzzle.java;

/**
 * Write a program that prints the numbers from 1 to 100 following the rules below:
 * - for multiples of three print “Fizz” (instead of the number)
 * - for the multiples of five print “Buzz”
 * - for numbers which are multiples of both three and five print “FizzBuzz”
 * - otherwise print the number
 */
public class FizzBuzz {

    public String fizzBuzz(int count) {
        StringBuilder sb = new StringBuilder();

        for (int i = 1; i <= count; i++) {
            sb.append((i % 3 == 0 && i % 5 == 0)
                    ? "FizzBuzz" : (i % 3 == 0)
                    ? "Fizz" : (i % 5 == 0)
                    ? "Buzz" : i).append(" ");
        }

        return sb.toString().trim();
    }
}
